README.md

```diff
@@ -93,9 +93,129 @@ TODO aspect ratio + images alignment + more more images
+
+### 🧩 Minimally Invasive Design
+[TODO]: vulkan handle acquisition, multiple windows, content playing second fiddle
+
+### 🧩 Designed for Interoperation
+Nabla is built with interoperation in mind, supporting memory export and import between different compute and graphics APIs.
+
+----
+
+[TODO]:
+- Cancellable Future based Async I/O
+- Virtual File System (archive mounting, our alternative to #embed, everything is referenced by absolute - path)
+
+----
+
+### 🧩 Asset Manager
+Nabla’s Asset Manager efficiently loads assets while tracking dependencies using a Directed Acyclic Graph (DAG). Assets are loaded in the correct order, avoids redundant allocations, and simplifies resource management.
+
+### 🧩 Asset Converter (CPU to GPU)
+The Asset Converter transforms CPU objects (asset::IAsset) into GPU objects (video::IBackendObject) while eliminating duplicates. Instead of relying on pointer comparisons, it hashes asset contents to detect and reuse identical GPU objects.
+
+### 🧩 Unit-Tested BxDFs for Physically Based Rendering
+A statically polymorphic library for defining Bidirectional Scattering Distribution Functions (BxDFs) in HLSL and C++. Each BxDF is rigorously unit-tested in C++ as well as HLSL. This is part of Nabla’s HLSL-C++ compatible library.
+
+### 🔧 In Progress: Property Pools (GPU Entity Component System)
+*Property Pools* group related properties together in a Structure Of Arrays (SoA) manner, allowing efficient, cache-friendly access to data on the GPU. The system enables transferring properties (Components) between the CPU and GPU, with the `PropertyPoolHandler` managing scattered updates with a special compute shader. Handles are assigned for each object and remain constant as data is added or removed.
+
+### 🧩 SPIR-V Introspection and Layout Creation
+
+SPIR-V introspection in Nabla eliminates most of the boilerplate code required to set up descriptor and pipeline layouts, simplifying resource binding to shaders.
+
+### 🧩 Nabla Extensions
+- ImGui integration.
+- Fast Fourier Transform for image processing and all kind of frequncy-domain fun.
+- Workgroup Prefix Sum – Efficient parallel prefix sum computation.
+- Blur – Optimized GPU-based image blurring.
+- Counting Sort – High-performance, GPU-accelerated sorting algorithm.
+- Autoexposure [Work in Progress] – Adaptive brightness adjustment for HDR rendering.
+- Tonemapping
+- GPU MPMC Queue – Multi-producer, multi-consumer GPU queue.
+- OptiX interoperability for ray tracing.
+- Global Scan – High-speed parallel scanning across large datasets.
+
+### 🚀 Coming Soon [TODO: Explain some better]
+- Full CUDA interoperability support.
+- Scene Loaders
+- GPU-Driven Scene Graph
+- Material Compiler 2.0 for efficient scheduling of BxDF graph evaluation
+
+### [TODO?] IUtiltities?  Using Fixed-sized staging memory for easier cpu-gpu transfers? format promotion?
```

Comment:
yeah that stuff can go to the top, above the Virtual File System

Reply:
Ah ok,I got my answer to the previous question. I was close :D

Reply:
done but not sure if I linked them correctly or tagged them as WIP correctly, please let me know if this part needs further modification